src/image.rs

```diff
@@ -0,0 +1,104 @@
+use crate::config::ListenerConfig;
+use std::sync::{Arc, RwLock};
+use byteorder::{ByteOrder, LittleEndian};
+use image::{ImageBuffer, DynamicImage, RgbaImage};
+
+use rosrust;
+use rosrust_msg;
+
+
+fn read_img_msg(img_msg: rosrust_msg::sensor_msgs::Image) -> DynamicImage {
+    let bytes = convert_u8(img_msg.data, &img_msg.encoding);
+    if img_msg.encoding.contains("rgb") {
+        return DynamicImage::ImageRgb8(ImageBuffer::from_raw(
+                img_msg.width,
+                img_msg.height,
+                bytes).unwrap())
+    } else {
+        return DynamicImage::ImageLuma8(ImageBuffer::from_raw(
+                img_msg.width,
+                img_msg.height,
+                bytes).unwrap())
+    }
+}
+
+fn convert_u8(vec: Vec<u8>, encoding: &str) -> Vec<u8> {
+    let mut bytes: Vec<u8> = Vec::new();
+    if encoding.contains("8") {
```

Comment:
Is there no other way? like a enum?
Maybe at least match to u8. This seems very hacky

Reply:
it is very hacky. i didnt want to re-implement http://docs.ros.org/en/jade/api/sensor_msgs/html/image__encodings_8h_source.html (my if contains 8 works for most types though) 
https://github.com/image-rs/image has some good ways for reading image types but i didnt manage to adapt it. 

this works for every common image type i encountered on robots. I would propose to merge 'shitty' encoding and open up a new  issue to do it in a sophisticated manner (likely in a different crate). The alternative would be to keep this open until proper conversion can be done.

Reply:
I need your opinions  (a crate would be even better)